pkg/agent/util/interfaces.go

```diff
@@ -0,0 +1,35 @@
+// Copyright 2023 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package util
+
+type Interface interface {
+	NewNetRoute(route *Route) error
+
+	RemoveNetRoute(route *Route) error
+
+	ReplaceNetRoute(route *Route) error
+
+	ReplaceNetNeighbor(neighbor *Neighbor) error
+
+	RouteListFiltered(family uint16, filter *Route, filterMask uint64) ([]Route, error)
+
+	ReplaceNetNatStaticMapping(mapping *NetNatStaticMapping) error
```

Comment:
NetNatStaticMapping is very Windows specific, it is a bit strange to work as a general Interface suitable for both Windows and Linux.